ktor-client/ktor-client-tests/common/test/io/ktor/client/tests/HttpRequestRetryTest.kt

```diff
@@ -56,6 +66,39 @@ class HttpRequestRetryTest {
         }
     }
 
+    @Test
+    fun testModifyRequest() = testWithEngine(MockEngine) {
+        config {
+            engine {
+                addHandler {
+                    assertEquals(null, it.headers["X-RETRY-COUNT"])
+                    respondError(HttpStatusCode.InternalServerError)
+                }
+                addHandler {
+                    assertEquals("1", it.headers["X-RETRY-COUNT"])
+                    respondError(HttpStatusCode.InternalServerError)
+                }
+                addHandler {
+                    assertEquals("2", it.headers["X-RETRY-COUNT"])
+                    respondError(HttpStatusCode.ServiceUnavailable)
+                }
+                addHandler {
+                    assertEquals("3", it.headers["X-RETRY-COUNT"])
+                    respondOk()
+                }
+            }
+            install(HttpRequestRetry) {
+                modifyRequest { it.headers.append("X-RETRY-COUNT", retryCount.toString()) }
```

Comment:
Should we make a retry handler with the request builder instead? I'm not sure that it's a most comfort way of modifying the request

Reply:
We can make a small design discussion about that